src/auto/device.rs

```diff
@@ -90,9 +91,33 @@ impl Device {
         unsafe { from_glib(gdk_sys::gdk_device_get_has_cursor(self.to_glib_none().0)) }
     }
 
-    //pub fn get_history<P: IsA<Window>>(&self, window: &P, start: u32, stop: u32, events: /*Ignored*/Vec<TimeCoord>) -> Option<i32> {
-    //    unsafe { TODO: call gdk_sys:gdk_device_get_history() }
-    //}
+    pub fn get_history<P: IsA<Window>>(
+        &self,
+        window: &P,
+        start: u32,
+        stop: u32,
+    ) -> Option<Vec<TimeCoord>> {
+        unsafe {
+            let mut events = ptr::null_mut();
+            let mut n_events = mem::MaybeUninit::uninit();
+            let ret = from_glib(gdk_sys::gdk_device_get_history(
+                self.to_glib_none().0,
+                window.as_ref().to_glib_none().0,
+                start,
+                stop,
+                &mut events,
+                n_events.as_mut_ptr(),
+            ));
+            if ret {
+                Some(FromGlibContainer::from_glib_full_num(
```

Comment:
Unhappy CI

```
error[E0277]: the trait bound `time_coord::TimeCoord: glib::translate::FromGlibContainerAsVec<_, *mut *mut gdk_sys::GdkTimeCoord>` is not satisfied

   --> src/auto/device.rs:188:22

    |

188 |                 Some(FromGlibContainer::from_glib_full_num(

    |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `glib::translate::FromGlibContainerAsVec<_, *mut *mut gdk_sys::GdkTimeCoord>` is not implemented for `time_coord::TimeCoord`

    |

    = note: required because of the requirements on the impl of `glib::translate::FromGlibContainer<_, *mut *mut gdk_sys::GdkTimeCoord>` for `std::vec::Vec<time_coord::TimeCoord>`

    = note: required by `glib::translate::FromGlibContainer::from_glib_full_num`
```